crates/ruff/src/rules/flake8_bugbear/rules/mutable_argument_default.rs

```diff
@@ -50,14 +53,23 @@ use crate::checkers::ast::Checker;
 pub struct MutableArgumentDefault;
 
 impl Violation for MutableArgumentDefault {
+    const AUTOFIX: AutofixKind = AutofixKind::Sometimes;
+
     #[derive_message_formats]
     fn message(&self) -> String {
         format!("Do not use mutable data structures for argument defaults")
     }
+    fn autofix_title(&self) -> Option<String> {
+        Some(format!("Replace mutable data structure with `None` in argument default and replace it with data structure inside the function if still `None`"))
+    }
 }
 
 /// B006
-pub(crate) fn mutable_argument_default(checker: &mut Checker, arguments: &Arguments) {
+pub(crate) fn mutable_argument_default(
+    checker: &mut Checker,
+    arguments: &Arguments,
+    body: &[Stmt],
```

Comment:
Nit: You could pass the whole FuncDef (or `AnyFuncDef`) instead of the individual arguments

Reply:
Hmm yeah that's complicated to do in isolation. You would need to have two dedicated branches, one for each function type, then call `AnyFunctionDef::from(func)`, and pass that `AnyFunctionDef` to a `check_any_function_def` that calls all existing rules. Let's leave it for now.